src/store/rating/actions.js

```diff
@@ -62,28 +62,47 @@ export async function fetchRatingsByTransaction ({ rootGetters }, { targetId, as
 
 export async function fetchRatedTransactions ({ commit, rootGetters }, { transactionsIds }) {
   const ratingsOptions = rootGetters.ratingsOptions
-  const nbEditableRatings = Object.keys(ratingsOptions.types).reduce((memo, key) => {
+
+  const ratingTypes = Object.keys(ratingsOptions.types)
+
+  const nbEditableRatings = ratingTypes.reduce((memo, key) => {
     if (!isRatingOptional(ratingsOptions.types[key])) memo += 1
     return memo
   }, 0)
 
-  const ratingsStats = await api.fetchRatingsStats({
-    transactionId: transactionsIds,
-    groupBy: 'transactionId'
-  })
-
-  const ratingsStatsByTransactionId = keyBy(ratingsStats, 'transactionId')
+  const fetchStats = ratingTypes.length > 1
 
   const ratedTransactions = {}
 
-  transactionsIds.forEach(transactionId => {
-    const stats = ratingsStatsByTransactionId[transactionId]
-    if (!stats) {
-      ratedTransactions[transactionId] = false
-    } else {
-      ratedTransactions[transactionId] = ratingsStatsByTransactionId[transactionId].count >= nbEditableRatings
-    }
-  })
+  if (fetchStats) {
```

Comment:
It seems you use this test here in rating store for `fetchRatedTransactions` (which is a strange name by the way, why not `fetchTransactionRatings`?) but not in other places where you’re fetching stats (as in inbox store, `fetchRatingsStats`).

Reply:
Indeed, I'll update the name.

This condition is only needed here because we want to retrieve the note per transaction, which is composed of multiple ratings only if there are multiple rating types. That's why depending on the number of rating types, the request will hit `list` or `stats` endpoint.

In other locations, we want to fetch the rating stats to have the average on all ratings.

Reply:
Thanks.